spec/prawn/images_spec.rb

```diff
@@ -43,11 +43,68 @@
     expect(info.height).to eq(453)
   end
 
+  it 'does not close passed-in IO objects' do
+    file = File.open(filename, 'rb')
+    _info = pdf.image(file)
+
+    expect(file).to_not be_closed
+  end
+
   it 'accepts Pathname objects' do
     info = pdf.image(Pathname.new(filename))
     expect(info.height).to eq(453)
   end
 
+  context 'closes opened files again after getting Pathnames', issue: 975 do
+    context 'spec with File message spy' do
+      let(:not_filename) { 'non-existent filename' }
+      let(:pathname_double) { instance_double('Pathname', file?: true) }
+
+      before do
+        file_content = File.new(filename, 'rb').read
+        allow(Pathname).to receive(:new).with(not_filename) { pathname_double }
+        allow(pathname_double).to receive(:binread) { file_content }
+      end
+
+      it 'uses only binread, which closes opened files' do
+        # this implicitly tests that a file handle is closed again
+        # because only stubbed binread can be called on not_filename
+        _info = pdf.image(not_filename)
+        expect(pathname_double).to have_received(:binread)
+      end
+    end
+
+    it 'closes opened files, spec with lsof' do
+      system_has_lsof = system('lsof -v > /dev/null 2>&1')
+      system_has_grep = system('grep --version > /dev/null 2>&1')
+      if system_has_lsof && system_has_grep
+        gc_was_disabled = GC.disable # GC of File would close the file
+        open_before = `lsof -c ruby | grep "#{filename}"`
+        _info = pdf.image(Pathname.new(filename))
+        open_after = `lsof -c ruby | grep "#{filename}"`
+        GC.enable unless gc_was_disabled
+        expect(open_after).to eq(open_before)
+      else
+        pending 'Platform dependent spec, only works on linux and macOS'
+        raise 'pending'
```

Comment:
Is this required? Wouldn't only previous line be enough?

Reply:
Rspec doesn't seem to like pending specs that don't fail. IIRC it was introduced in Rspec 3:

<img width="983" alt="screen shot 2018-11-26 at 11 53 19" src="https://user-images.githubusercontent.com/102580/49009911-0ce99300-f172-11e8-92ea-ce14c4323b16.png">

Error: `Expected pending 'Platform dependent spec, only works on linux and macOS' to fail. No error was raised.`